pkgs/applications/virtualization/xen/patches.nix

```diff
@@ -0,0 +1,63 @@
+# Patching Xen? Check the XSAs at
+# https://xenbits.xen.org/xsa/
+# and try applying all the ones
+# we haven't gotten around to
+# yet, if any are necessary.
+
+{ fetchpatch }:
+
+let
+  xsaPatch =
+    { name, hash }:
+    (fetchpatch {
+      url = "https://xenbits.xen.org/xsa/xsa${name}.patch";
+      inherit hash;
+    });
+  qubesPatch =
+    {
+      name,
+      tag,
+      hash,
+    }:
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/QubesOS/qubes-vmm-xen/v${tag}/${name}.patch";
+      inherit hash;
+    });
+in
+{
+  # Example patches:
+  #
+  # "XSA_100" = xsaPatch {
+  #   name = "100";
+  #   hash = "sha256-0000000000000000000000000000000000000000000000000000";
+  # };
+  #
+  # "QUBES_libxl-fix-all-issues" = qubesPatch {
+  #   name = "1000-libxl-fix-all-issues";
+  #   tag = "4.20.0-1";
+  #   hash = "sha256-0000000000000000000000000000000000000000000000000000";
+  # };
+
+  # Build reproducibility patches for Xen 4.17.4, but works on 4.18.2.
+  QUBES_Define-build-dates-time-based-on-SOURCE_DATE_EPOCH = qubesPatch {
+    name = "1100-Define-build-dates-time-based-on-SOURCE_DATE_EPOCH";
+    tag = "4.17.4-3";
+    hash = "sha256-OwKA9oPTwhRcSmiOb+PxzifbO/IG8IHWlvddFh/nP6s=";
+  };
+  QUBES_docs-rename-DATE-to-PANDOC_REL_DATE-and-allow-to-spe = qubesPatch {
+    name = "1101-docs-rename-DATE-to-PANDOC_REL_DATE-and-allow-to-spe";
+    tag = "4.17.4-3";
+    hash = "sha256-BUtYt0mM3bURVaGv4oDznzxx1Wo4sfOpGV5GB8qc5Ns=";
+  };
+  QUBES_docs-xen-headers-use-alphabetical-sorting-for-incont = qubesPatch {
+    name = "1102-docs-xen-headers-use-alphabetical-sorting-for-incont";
+    tag = "4.17.4-3";
+    hash = "sha256-mQUp2w9lUb7KDq5MuPQjs6y7iuMDeXoZjDjlXfa5z44=";
+  };
```

Comment:
Should probably be tagged to the current version instead of reusing the old once

Reply:
This is the current version. Qubes has not updated to Xen 4.18/4.19 yet.

Reply:
See https://github.com/NixOS/nixpkgs/pull/324693/files/c9796c05ee1c13ca9f9fd997248862a74be2a1b0#r1685830201